packages/config-dug/src/config-dug.ts

```diff
@@ -248,20 +247,31 @@ class ConfigDug<T extends ConfigDugSchema> extends EventEmitter {
   private async loadPlugins(): Promise<UntypedConfig> {
     let values: UntypedConfig = {};
 
+    const { minReloadInterval, reloadInterval } = this.options;
+
+    let nextReloadIn: number = reloadInterval;
+
     for (const plugin of this.options.plugins) {
       const pluginReturnValue: ConfigDugPluginOutput = await plugin.load();
 
       values = { ...values, ...pluginReturnValue.values };
 
       this.valueOrigins = mergeOrigins(this.valueOrigins, pluginReturnValue.valueOrigins);
 
-      if (pluginReturnValue.nextReloadIn) {
-        this.reloadTimeout = setTimeout(async () => {
-          await this.reload();
-        }, pluginReturnValue.nextReloadIn);
+      const pluginNextReloadIn = pluginReturnValue.nextReloadIn;
+
+      if (pluginNextReloadIn && pluginNextReloadIn < nextReloadIn) {
+        nextReloadIn = pluginNextReloadIn;
       }
     }
 
+    setTimeout(
```

Comment:
This looks good but it seems like it should exist on the top level object, not in the `loadPlugins` function

Reply:
The reason I added here was because I didn't want to loop through the `this.options.plugins` array again in that top level function to get the shortest reload interval for plugins. Maybe I can return from the `loadPlugins` function the configs and also the shortest reload interval to be used on the main `loadConfig` function.